Content.Server/Chemistry/ReagentEffects/HealthChange.cs

```diff
@@ -27,14 +26,10 @@ public class HealthChange : ReagentEffect
         /// Damage group to change.
         /// </summary>
         // TODO PROTOTYPE Replace this datafield variable with prototype references, once they are supported.
-        // Also requires replacing DamageGroup() calls with something like _damageGroup
         [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
         [DataField("damageGroup", required: true)]
         private readonly string _damageGroupID = default!;
-        private DamageGroupPrototype DamageGroup() {
-            IoCManager.InjectDependencies(this);
-            return _prototypeManager.Index<DamageGroupPrototype>(_damageGroupID);
-        }
+        private DamageGroupPrototype DamageGroup => IoCManager.Resolve<IPrototypeManager>().Index<DamageGroupPrototype>(_damageGroupID);
```

Comment:
Depending on how often this is called, it might be better to go with the [Dependency]. If it isn't called that much then remember to delete line 29.

Reply:
I've changed around all of the "TODO PROTOTYPE" datafield blocks. Now instead of using " => " to get prototypes, wherever possible the prototypes just get set when first initialised. It's what I should have done from the begining.

Health change is slightly different from most instances, because its not a component with an Initialize() function, so I used ISerializationHooks.AfterDeserialization instead. Hopefully that's the correct thing to do.